app/controllers/sign_up/completions_controller.rb

```diff
@@ -15,6 +15,7 @@ def show
       )
       @multiple_factors_enabled = MfaPolicy.new(current_user).multiple_factors_enabled?
       @presenter = completions_presenter
+      user_session[:pending_completions_consent] = true
```

Comment:
Separately, I'd wondered if it would be enough to just check [`needs_completion_screen_reason`](https://github.com/18F/identity-idp/blob/74b19c4b4ab14ca9eaf1eb70171b0e857018d85c/app/controllers/concerns/verify_sp_attributes_concern.rb#L4) to know if the user is pending consent with the partner, rather than adding a new session value.

But I could imagine some cases where the user could manage their emails without having seen the consent screen, while still having a value for `needs_completion_screen_reason`. Though the same could be said of this session value.

Curious if you have any thoughts.

Reply:
It looks like it does work the same. Seems like the session isn't needed?

Reply:
The main difference is that "needs completion screen reason" is going to be `true` if the user needs to consent with the partner, regardless whether they've actually seen the consent screen. But the value controls whether the user sees the consent screen, so usually they're closely linked. There's very few cases I could imagine where a user could find themselves managing their account emails while having a completion screen reason, and the same edge-cases would apply even if we used a session value. So it could simplify things.

Reply:
Removed that session token.